### ISSUE TITLE
Bring `weight` back in header for sort order on All Pages side menu

### ISSUE THREAD

**Issue by joewheaton** [Riverscapes/riverscapes-gatsby-theme#33]
I am really hating that this went away on migration from Jekyll to Gatsby. This is a major problem not having control over the sort order. It doesn't even seem to be alpha-numeric? I would love to not have to put numbers on these. The order does matter on these auto-populated site menus! It currently appears to be doing it based on last modified date? That is absurd.

![image](https://github.com/user-attachments/assets/ba38289f-68a2-4791-a400-463d02c4be36)

This was far superior design, function, layout and had ability with weight to control sort order:
![image](https://github.com/user-attachments/assets/5b81f1c2-361c-4c77-8d9c-a4281a1c9917)

Bring the equivalent functionality back please!

**Comment by MattReimer** [comment]
Yeah. mandatory numbering is dumb. We can bring `weight` back but it should be optional and pages without it should still sort alphabetical for simplicity.

**Comment by MattReimer** [comment]
Weights are now back in. If you include `menuWeight: 3` in the frontmatter then that will be used. 

Menu items are sorted first by weight and second by name (alphabetically). Items without weights are given 999 as a default so they fall to the bottom.
